lib/js/components/DatePickers/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,278 @@
+				:eyebrow-text="eyebrowText"
+				@click="toggle"
+			/>
+		</template>
+		<date-picker-box
+			v-else
+			:is-interactive="isInteractive"
+			:placeholder="placeholder"
+			:date="date"
+			:icon="icon"
+			:is-icon-hidden-on-mobile="isIconHiddenOnMobile"
+			:state="state"
+			:color="color"
+			:eyebrow-text="eyebrowText"
+			:is-open="isOpen"
+			@click="toggle"
+		/>
+
+		<span class="ds-datePicker__errorMessage"> {{ errorMessage }} </span>
+		<input ref="flatpickrInput" class="ds-datePicker__hiddenInput" />
+	</div>
+</template>
+
+<style lang="scss">
+@import '../../../../styles/settings/typography/tokens';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/spacings';
+@import 'flatpickr/dist/flatpickr';
+
+.ds-datePicker {
+	$self: &;
+
+	display: flex;
+	flex-direction: column;
+	row-gap: $space-4xs;
+
+	&.-ds-box {
+		display: inline-flex;
+	}
+
+	&__hiddenInput {
+		display: none;
+	}
+
+	&__label {
+		@include label-l-default-bold;
+
+		color: $color-neutral-text-strong;
+		padding-bottom: $space-4xs;
+
+		&.-ds-isUppercase {
+			@include label-m-extensive-bold-uppercase;
+
+			color: $color-neutral-text;
+		}
+
+		&.-ds-disabled {
+			color: $color-neutral-text-strong-disabled;
+
+			&.-ds-isUppercase {
+				color: $color-neutral-text-disabled;
+			}
+		}
+	}
+
+	&__errorMessage {
+		@include info-s-default-regular;
+
+		color: $color-danger-text;
+	}
+}
+</style>
+
+<script lang="ts">
+import { toRaw, PropType } from 'vue';
+
+import DsTile from '../../Tile';
+import { ICONS } from '../../Icons/Icon';
+import DatePickerBox from '../DatePickerBox';
+
+import {
+	DATE_PICKER_CALENDAR_POSITIONS,
+	DATE_PICKER_COLORS,
+	DATE_PICKER_STATES,
+	DATE_PICKER_TRIGGER_TYPES,
+	DatePickerCalendarPositions,
+	DatePickerColors,
+	DatePickerStates,
+	DatePickerTriggerTypes,
+} from './DatePicker.consts';
+import { capitalizeFirstLetter } from '../../../../../tools/importers/helpers/modifiers';
+
+let flatpickr: Function | null = null;
+
+const flatpickrPositions = {
+	[DATE_PICKER_CALENDAR_POSITIONS.BOTTOM_LEFT]: 'below left',
+	[DATE_PICKER_CALENDAR_POSITIONS.BOTTOM_RIGHT]: 'below right',
+	[DATE_PICKER_CALENDAR_POSITIONS.BOTTOM]: 'below',
+	[DATE_PICKER_CALENDAR_POSITIONS.TOP_LEFT]: 'above left',
+	[DATE_PICKER_CALENDAR_POSITIONS.TOP_RIGHT]: 'above right',
+	[DATE_PICKER_CALENDAR_POSITIONS.TOP]: 'above',
+};
+export default {
+	name: 'DatePicker',
+	components: {
+		DatePickerBox,
+		DsTile,
+	},
+	props: {
+		triggerType: {
+			type: String as PropType<DatePickerTriggerTypes>,
+			default: DATE_PICKER_TRIGGER_TYPES.TILE,
+		},
+		isInteractive: {
+			type: Boolean,
+			default: true,
+		},
+		placeholder: {
+			type: String,
+			default: 'Wybierz datę',
+		},
+		date: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		label: {
+			type: String,
+			default: '',
+		},
+		isLabelUppercase: {
+			type: Boolean,
+			default: false,
+		},
+		icon: {
+			type: Object,
+			default: null,
```

Comment:
in mirro we have a `fa-icon-calendar-day` as default, https://miro.com/app/board/o9J_knPnexA=/?moveToWidget=3458764563627946445&cot=14

Reply:
I set it default in story but setting the default in component makes it impossible to be null thus cannot hide an icon

Reply:
then you are doing something wrong, because I easily set it as default and can use null :P

Reply:
```
icon: {
			type: [Object, null],
			default: ICONS.FA_CALENDAR_DAY,
},
```

and in storybook
```
:icon="icon ? ICONS[icon] : null"
```

Reply:
yeah, probably, thx